src/components/primitives/Forms/NumericalInput.tsx

```diff
@@ -1,22 +1,34 @@
 // Based on https://github.com/levelkdev/dxswap-dapp/blob/master/src/components/Input/NumericalInput/index.tsx
 
-import React from 'react';
+import React, { useState } from 'react';
 import { Input, InputProps } from 'components/primitives/Forms/Input';
 import { escapeRegExp } from 'utils';
+import { IconRight } from './IconRight';
 
 const inputRegex = RegExp(`^\\d*(?:\\\\[.])?\\d*$`); // match escaped "." characters via in a non-capturing group
 
 export const NumericalInput: React.FC<InputProps<string>> = ({
   value,
   onChange,
   placeholder,
+  disabled = true,

```

Comment:
I think passing defaultValue here would be more flexible in future and would limit the locking to when it is explicitly needed